### PR TITLE
Add option to sort by single and average in psych sheets - Fixes #295

### LINKED ISSUE #295: Option to sort by single and average in psych sheets

### PR DESCRIPTION
**Pull request by pingskills**
By default this sorts by the preferred format of the event. I've then added a small amount of text to show how this is sorted, with a link to sort by the other option. Here are some screenshots. Let me know what you think.

sorted-by-single

sorted-by-average

### CODE REVIEW

**Comment by jfly** [comment]
(I haven't looked at the code yet)

Could we instead style this like our other tables and have little triangles and make the table headers clickable? Like what we do on https://www.worldcubeassociation.org/users.

**Comment by pingskills** [comment]
Hi @jfly,

We could do that but then the first column would not be correct. Maybe that's OK? See the screenshots, if we just sorted the table then he would be number 1 in both sorts, even though he is number 2 when sorted by single.

**Comment by jfly** [comment]
Ah, I see. Whatever you prefer, then =)

File: WcaOnRails/app/models/competition.rb [hunk 1]
```diff
@@ -699,15 +699,15 @@ def psych_sheet_event(event)
 
     prev_registration = nil
     registrations.each_with_index do |registration, i|
-      if event.recommended_format.sort_by == :single
+      if sort_by == 'single'
```
Comment (jfly) [inline WcaOnRails/app/models/competition.rb]: Changing this from a symbol to a string doesn't seem like it would be without side effects... how does this still work?

File: WcaOnRails/app/models/competition.rb [hunk 2]
```diff
@@ -699,15 +699,15 @@ def psych_sheet_event(event)
...
         rank = registration.single_rank
         prev_rank = prev_registration&.single_rank
       else
         rank = registration.average_rank
         prev_rank = prev_registration&.average_rank
       end
-      registration.tied_previous = (rank == prev_rank)
       break if !rank # hasn't competed in this event yet and all subsequent registrations too
+      registration.tied_previous = (rank == prev_rank)
```
Comment (jfly) [inline WcaOnRails/app/models/competition.rb]: This seems like a subtle change. Is this related to the feature, or something else?

File: WcaOnRails/app/models/competition.rb (hunk 1 above)
Comment (pingskills) [inline WcaOnRails/app/models/competition.rb]: It actually didn't work before. This part of the code is used to tell if the already sorted competitors are tied. The `event.recommended_format.sort_by == :single` always failed and hence all ties were computed by the average_rank (although the sorting was already performed correctly). The main test looking at ties was done using average. When I added a test for single it failed. So I've now changed the comparison and it now works.

File: WcaOnRails/app/models/competition.rb (hunk 2 above)
Comment (pingskills) [inline WcaOnRails/app/models/competition.rb]: I didn't really need to do this. But I thought it was a more appropriate place to break. Once users no longer have a rank, we stop giving them a position, so we should also stop seeing if they are tied.

**Comment by jfly** [comment]
Ah, that's too bad we didn't have a test for this. I believe this used to
be a symbol, but would have broke when you moved this stuff into the
database and everything became strings :) Thanks for adding a test.

**Comment by jfly** [comment]
OK, I could go both ways here. I think it's unfortunate that tied_previous
is no longer just true or false, but so long as we test this, I'm fine with
this change.

**Comment by jfly** [comment]
LGTM!
